sections/docker/Caching-for-shorter-build-time.md

```diff
@@ -0,0 +1,120 @@
+# Leverage caching to reduce build times
+
+## One paragraph explainer
+
+Docker images are a combination of layers, each instruction in your Dockerfile creates a layer. The docker daemon can reuse those layers between builds if the instructions are identicals or in the case of a `COPY` or `ADD` files used are identical.
+It is important to layout your Dockerfile correctly to reduce the number of moving parts in your build, the less updated instructions should be at the top and the ones constantly changing (like app code) should be at the bottom. 
+Rebuilding a whole docker image from cache can be nearly instantaneous if done correctly.
+
+⚠️ If the cache can't be used for a particular layer, all the subsquent layers will be invalidated too. That's why order is important.
+
+![Docker layers](/assests/images/docker_layers_schema.png)
+
+* Image taken from [Digging into Docker layers](https://medium.com/@jessgreb01/digging-into-docker-layers-c22f948ed612) by jessgreb01*
+
+### Rules
```

Comment:
This deviates a bit from our formatting (having custom section that is no code example or quote) but let's stick to it - it's good. For future bullets, I would stick to the formatting